mnefun/_mnefun.py

```diff
@@ -2750,6 +2802,45 @@ def _prebad(p, subj):
     return prebad_file
 
 
+def _maxbad(p, subj, raw_fname, bad_file):
+    """Helper for handling Maxfilter bad channels selection prior to SSS"""
+    prebad_file = _prebad(p, subj)
+    if not op.isfile(bad_file):
+        print('         Could not find MaxFilter bad file: %s' %
+              bad_file + ' ... Running on kasga.\n', end='')
+        opts = '-v -force'
+        if not p.mf_badlimit == 7:
+            mf_kwargs = {'badlimit': p.mf_badlimit}
+        run_sss_command(raw_fname, opts, None, host=p.sws_ssh,
+                        fname_bads=bad_file,
+                        work_dir=p.sws_dir, **mf_kwargs)
+        with open(bad_file, "r") as f:
+            output = [line for line in f]
+        # Parse output for bad channels
+        if any('Static bad channels' in ss for ss in output):
+            chs = [ll for ll in output if 'Static bad channels' in ll][
+                0].split(':')[1].strip()
+            with open(bad_file, "w+") as f:
+                f.write(chs)
+        print('Backing-up %s.orig: ' % prebad_file,
+              '\nWriting modified %s: ' % prebad_file, end='')
+        copy2(prebad_file, prebad_file + '.orig')
```

Comment:
I would not overwrite anything, but instead when we want to use the prebad/maxbad files (always together, probably only in the Maxfiltering step):

1. Read the channels from the manually created `_prebad.txt` file (if it exists)
2. Read the channels from the MF-created `_maxbad.txt` file
3. Concat these in memory like `all_bads = sorted(set(prebads) + set(maxbads))`

Then use `raw.info['bads'] = all_bads` or so

Reply:
It looks like this has not been addressed

Reply:
I’ll get to it today